### PR TITLE
Annual income range filter

### PR DESCRIPTION
**Pull request by bhawanabadlani**
Added annual income range filter to filter the dataset according to annual_income. This range filter **accepts amount along with currency, separated by a space**. The currency feature add-ons to the filter for it helps in querying the annual income with different currencies.
image

### CODE REVIEW

**Comment by rtnpro** [comment]
Nice progress. The filter is working with a few issues:

- Filter values does not get populated after filtering
image

- It breaks if I do not enter currency. It should not break.

image

File: vmb/matrimony/admin.py [hunk 1]
```diff
@@ -70,9 +72,107 @@ def queryset(self, request, queryset):
                 rounds_chanting__lt=int(8), rounds_chanting__gte=int(1)
             )
         if self.value() == "0":
+
             return queryset.filter(rounds_chanting__lte=int(0))
 
 
+class InputFilter(admin.SimpleListFilter):
```
Comment (rtnpro) [inline vmb/matrimony/admin.py]: Why is this required? Seems like from some source code. Can't we import it?

File: vmb/matrimony/admin.py [hunk 2]
```diff
@@ -70,9 +72,107 @@ def queryset(self, request, queryset):
...
+            self.used_parameters[self.parameter_name + "_to"] = to_currency
+
+    def queryset(self, request, queryset):
+        filters = {}
+
+        value_from = self.used_parameters.get(self.field_name + "_from", None)
+        currency_from = self.used_parameters.get(self.parameter_name + "_from", None)
+        if value_from is not None and value_from != "":
+            filters.update(
+                {self.field_name + "__gte": Money(value_from, currency_from),}
```
Comment (rtnpro) [inline vmb/matrimony/admin.py]: Nice!

File: vmb/matrimony/admin.py [hunk 3]
```diff
@@ -70,9 +72,107 @@ def queryset(self, request, queryset):
...
+
+        if self.parameter_name + "_from" in params:
+            value = (params.pop(self.parameter_name + "_from")).split()
+            from_income = value[0]
+            from_currency = value[1]
+            self.used_parameters[self.field_name + "_from"] = from_income
+            self.used_parameters[self.parameter_name + "_from"] = from_currency
+
+        if self.parameter_name + "_to" in params:
+            value = (params.pop(self.parameter_name + "_to")).split()
```
Comment (rtnpro) [inline vmb/matrimony/admin.py]: ~You should not modify/pop `self.parameter_name + "_to"` or `_from`. That's why the applied filter values are not getting rendered in the `from` to `to` fields post filtering.~

**Review by rtnpro — CHANGES_REQUESTED** [review]

File: vmb/matrimony/admin.py (hunk 3 above)
Comment (rtnpro) [reply vmb/matrimony/admin.py]: This is the issue.

image

Comment (rtnpro) [reply vmb/matrimony/admin.py]: Check this here:
https://github.com/jpsoffice/vmb/blob/master/vmb/matrimony/admin.py#L93

This value gets rendered in the filter form field.

But for querying, I am using `self.field_name + '_from'`, https://github.com/jpsoffice/vmb/blob/master/vmb/matrimony/admin.py#L107

Comment (bhawanabadlani) [reply vmb/matrimony/admin.py]: Ok. Thank you for the detailed review. Will make the necessary changes.

**Comment by bhawanabadlani** [comment]
Resolved the issue: breaking of code when currency is not entered. Along with it, invalid currency, as well as invalid amount format, is also taken care of. Please review.

**Comment by rtnpro** [comment]
@bhawanabadlani  Filter is not filtering properly

image

**Review by rtnpro — APPROVED** [review]
